### PR TITLE
DT-235 UI summary: Use DepartureTime and RouteNumber components

### PR DESCRIPTION
**Pull request by Macroz**
[DT-235 UI summary: Use DepartureTime and RouteNumber components](https://digitransit.atlassian.net/browse/DT-235)

Also includes some minor component refactorings (from component-refactor)

### CODE REVIEW

File: app/component/summary/summary-row.cjsx [hunk 1]
```diff
@@ -45,15 +48,13 @@ class SummaryRow extends React.Component
         position: 'absolute'
         left: "calc(((100% - (#{data.legs.length} * #{MIN_SIZE})) * #{position}) + (#{i} * #{MIN_SIZE}))"
         overflow: 'hidden'
-        color: 'black'
-        'fontWeight': 400
+        fontWeight: 400
```
Comment (hannesj) [inline app/component/summary/summary-row.cjsx]: Should the static styles be extracted to css?

File: app/component/summary/summary-row.cjsx [hunk 2]
```diff
@@ -79,39 +80,31 @@ class SummaryRow extends React.Component
...
-        <span key={i + 'a'} style={styleLine} className={leg.mode.toLowerCase()}>
-          <span key={i + 'b'} className="summary-circle #{circleClass}#{passiveClass}"></span>
-          <Icon key={i + 'c'} className={leg.mode.toLowerCase()} img={'icon-icon_' + leg.mode.toLowerCase()} />
-          {text}
-        </span>
-      )
+      legs.push <span key={i + 'a'}
+        style={styleLine}
+        className={leg.mode.toLowerCase()}>
+        <span key={i + 'b'} className="summary-circle #{circleClass}#{passiveClass}"></span>
```
Comment (hannesj) [inline app/component/summary/summary-row.cjsx]: use classNames to concatenate class names?

File: app/component/summary/summary-row.cjsx [hunk 3]
```diff
@@ -79,39 +80,31 @@ class SummaryRow extends React.Component
...
+        currentTime={currentTime}
+        style={styleTime} />
 
       if isFirstLeg
-        legTimes.push (
-          <span key={i + 'a'} style={styleTime}>
-            {legStart.format("HH:mm")}
-          </span>
-        )
+        legTimes.push startTimeComponent
```
Comment (hannesj) [inline app/component/summary/summary-row.cjsx]: Could all of these be put inside something like unless isLastLeg and not isEnoughRoomForLastLegStartTime and separate check for is isLastLeg to put the arrival time for the journey.

File: app/component/summary/summary-row.cjsx (hunk 1 above)
Comment (Macroz) [inline app/component/summary/summary-row.cjsx]: Yes, makes sense to refactor more

**Pull request merged**